packages/contact-center/store/tests/store.test.ts

```diff
@@ -0,0 +1,104 @@
+import { makeAutoObservable } from 'mobx';
+import Webex from 'webex';
+import store from '../src/store'; // Adjust the import path as necessary
+
+jest.mock('mobx', () => ({
+  makeAutoObservable: jest.fn(),
+  observable: { ref: jest.fn() }
+}));
+
+jest.mock('webex', () => ({
+  init: jest.fn(() => ({
+    once: jest.fn((event, callback) => {
+      if (event === 'ready') callback();
+    }),
+    cc: {
+      register: jest.fn()
+    }
+  }))
+}));
+
+describe('Store', () => {
+  let mockWebex;
+
+  beforeEach(() => {
+    // Reset teams and loginOptions before each test since store is a singleton
+    store.teams = [];
+    store.loginOptions = [];
+    mockWebex = {
+      cc: {
+        register: jest.fn()
+      }
+    };
+  });
+
+  it('should initialize with default values', () => {
+    expect(store.teams).toEqual([]);
+    expect(store.loginOptions).toEqual([]);
+    expect(makeAutoObservable).toHaveBeenCalledWith(store, { cc: expect.any(Function) });
+  });
+
+  describe('registerCC', () => {
+    it('should set teams and loginOptions on successful register', async () => {
+      const mockResponse = {
+        teams: [{ id: 'team1', name: 'Team 1' }],
+        loginVoiceOptions: ['option1', 'option2']
+      };
+      mockWebex.cc.register.mockResolvedValue(mockResponse);
+
+      await store.registerCC(mockWebex);
+
+      expect(store.teams).toEqual(mockResponse.teams);
+      expect(store.loginOptions).toEqual(mockResponse.loginVoiceOptions);
+    });
+
+    it('should log an error on failed register', async () => {
+      const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation();
+      const mockError = new Error('Register failed');
+      mockWebex.cc.register.mockRejectedValue(mockError);
+
+      await store.registerCC(mockWebex);
+
+      expect(consoleErrorSpy).toHaveBeenCalledWith('Error registering contact center', mockError);
+      consoleErrorSpy.mockRestore();
+    });
+  });
+
+  describe('init', () => {
+    it('should call registerCC if webex is in options', async () => {
+      const initParams = { webex: mockWebex };
+      jest.spyOn(store, 'registerCC').mockResolvedValue();
+
+      await store.init(initParams);
+
+      expect(store.registerCC).toHaveBeenCalledWith(mockWebex);
```

Comment:
Can we add a check to ensure that it doesn't call, Webex.init in this case

Reply:
Done